tests/test_config.py

```diff
@@ -0,0 +1,116 @@
+from textwrap import dedent
+
+from vulture.config import Config, _parse_args, _parse_toml, make_config
+
+
+def test_cli_args():
+    """
+    Ensure that CLI arguments are converted to a config object.
+    """
+    expected = dict(
+        paths=["path1", "path2"],
+        exclude=["file*.py", "dir/"],
+        ignore_decorators=["deco1", "deco2"],
+        ignore_names=["name1", "name2"],
+        make_whitelist=True,
+        min_confidence=10,
+        sort_by_size=True,
+        verbose=True,
+    )
+    result = _parse_args(
+        [
+            "--exclude=file*.py,dir/",
+            "--ignore-decorators=deco1,deco2",
+            "--ignore-names=name1,name2",
+            "--make-whitelist",
+            "--min-confidence=10",
+            "--sort-by-size",
+            "--verbose",
+            "path1",
+            "path2",
+        ]
+    )
+    assert isinstance(result, Config)
+    assert result == expected
+
+
+def test_toml_config():
+    """
+    Ensure parsing of TOML files results in a valid config object.
+    """
+    expected = dict(
+        paths=["path1", "path2"],
+        exclude=["file*.py", "dir/"],
+        ignore_decorators=["deco1", "deco2"],
+        ignore_names=["name1", "name2"],
+        make_whitelist=True,
+        min_confidence=10,
+        sort_by_size=True,
+        verbose=True,
+    )
+    data = StringIO(
+        dedent(
+            u"""\
+        [tool.vulture]
+        exclude = ["file*.py", "dir/"]
+        ignore_decorators = ["deco1", "deco2"]
+        ignore_names = ["name1", "name2"]
+        make_whitelist = true
+        min_confidence = 10
+        sort_by_size = true
+        verbose = true
+        paths = ["path1", "path2"]
+        """
+        )
+    )
+    result = _parse_toml(data)
+    assert isinstance(result, Config)
+    assert result == expected
+
+
+def test_config_merging():
+    """
+    If we have both CLI args and a ``pyproject.toml`` file, the CLI args should
+    have precedence.
+    """
+    toml = StringIO(
+        dedent(
+            u"""\
+        [tool.vulture]
+        exclude = ["toml_exclude"]
+        ignore_decorators = ["toml_deco"]
+        ignore_names = ["toml_name"]
+        make_whitelist = false
+        min_confidence = 10
+        sort_by_size = false
+        verbose = false
+        paths = ["toml_path"]
+        """
+        )
+    )
+    cliargs = [
+        "--exclude=cli_exclude",
+        "--ignore-decorators=cli_deco",
+        "--ignore-names=cli_name",
+        "--make-whitelist",
+        "--min-confidence=20",
+        "--sort-by-size",
+        "--verbose",
+        "cli_path",
+    ]
+    result = make_config(cliargs, toml)
+    expected = dict(
+        paths=["cli_path"],
+        exclude=["cli_exclude"],
+        ignore_decorators=["cli_deco"],
+        ignore_names=["cli_name"],
+        make_whitelist=True,
+        min_confidence=20,
+        sort_by_size=True,
+        verbose=True,
+    )
+    assert result == expected
```

Comment:
Thanks for adding the test. It works, because all values are set in both CLI and TOML. However, I think it will fail if I add `verbose = true` to the TOML file, but don't pass `--verbose` on the command line. We need to test this scenario, in this test, or a separate test, whatever you prefer.

Reply:
Should be fine now. See ff30697

The is a slight issue with merging these values which is why I was forced to use a sentinel value to detect unset values. Otherwise the boolean checks would be difficult to write (if not impossible).

Let me know what you think.

Reply:
Using a sentinel like is a habit of mine, which I use in these situations. Mainly for consistency because sometimes you want to also flag something as having "no default set" while still being able to store the value "None". And if I use `None` to flag something as having "no default" I can no longer determine whether the value was supposed to be `None` or whether I should use the default value. It also keeps things more explicit in code. Reading something like `if valus is NO_DEFAULT` communicates the intent better than reading `if value is None` (IMO). 

Considering that all the values here are booleans, we could use `None` as this special marker.

Let me know what you think. If I will read an imperative sentence in the discussion, I will follow the order :smile:

Reply:
Point taken. Let's keep the special sentinel value :-)